test/postgresql.autoupdate.test.js

```diff
@@ -677,4 +679,274 @@ describe('autoupdate', function() {
       });
     });
   });
+
+  describe('multicolumn indexes on table in schema', () => {
+    const schema = {
+      name: 'Person',
+      options: {
+        postgresql: {
+          schema: 'public',
+          table: 'Person',
+        },
+        indexes: {
+          uniqueNameIndex: {
+            keys: {'firstName': 1, 'lastName': 1},
+            options: {unique: false},
+          },
+        },
+      },
+      properties: {
+        firstName: {
+          type: 'string',
+        },
+        lastName: {
+          type: 'string',
+        },
+        middleName: {
+          type: 'string',
+        },
+      },
+    };
+    const changedSchema = {
+      ...schema,
+      options: {
+        ...schema.options,
+        indexes: {
+          uniqueNameIndex: {
+            keys: {'firstName': -1, 'lastName': -1},
+            options: {unique: true},
+          },
+        },
+      },
+    };
+    const DROP_INDEX_REGEX = /DROP INDEX.*uniqueNameIndex/;
+
+    let personModel;
+
+    afterEach(async () => {
+      await personModel.destroyAll();
+    });
+
+    describe('when table has no existing indexes', () => {
+      it('updates without errors', async () => {
+        personModel = ds.define(schema.name, schema.properties, schema.options);
+        await ds.autoupdate([schema.name]);
+        assert(true, true);
+      });
+    });
+
+    describe('when table has existing index', () => {
+      let sandbox;
+      let spy;
+
+      before(() => { sandbox = sinon.createSandbox(); });
+      after(() => { sandbox = null; });
+
+      beforeEach(setupExistingSchemaAndSpy);
+      afterEach(teardownSpy);
+
+      it('removes existing index successfully', (done) => {
+        const noIndexSchema = {
+          ...schema,
+          options: {
+            ...schema.options,
+            indexes: {},
+          },
+        };
+
+        async.series([
```

Comment:
Let's rewrite this test from callback style to `await` style please.

```js
it('removes existing index successfully', async () => {
  // ...
  await ds.autoupdate([schema.name]);
  // ... 
  const indexes = ds.connector.discoverModelIndexes(schema.name);
  // ...
});
```

If the method `discoverModelIndexes` does not support promises yet, then you can add support for promises. See https://github.com/strongloop/loopback-connector-couchdb2/pull/83 for inspiration, I think we don't need a dedicated test for this change, since the method is not considered as a public API. Alternatively, you can add a small wrapper to this test file:

```js
function discoverModelIndexes(ds, schemaName) {
  return new Promise((resolve, reject) => {
    ds.connector.discoverModelIndexes(schemaName, (err, indexes) => {
       if (err) reject(err) else resolve(indexes);
    });
}
```

Reply:
@bajtos Yes, I'll make the change. `disoverModelIndexes` doesn't support promises, so that's why I was using the callback.

Reply:
@bajtos Yes, I'll make that change to add support for promises. `discoverModelIndexes` doesn't yet support promises, so that's why I was using callbacks.